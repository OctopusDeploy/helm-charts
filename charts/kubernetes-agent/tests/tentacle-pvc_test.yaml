suite: "persistence"
templates:
- templates/tentacle-pvc.yaml
tests:
- it: "should match snapshot when storageClassName is set"
  set:
    persistence:
      storageClassName: "my-storage-class"
  asserts:
    - matchSnapshot: {}
- it: "is not created when storageClassName is empty"
  set:
    persistence:
      storageClassName: ""
  asserts:
  - hasDocuments:
      count: 0

- it: "is created when storageClassName has a value"
  set:
    persistence:
      storageClassName: "my-storage-class"
  asserts:
  - hasDocuments:
      count: 1

- it: "can have the size changed"

  set:
    persistence:
      storageClassName: "my-storage-class"
      size: 16Gi
  asserts:
  - hasDocuments:
      count: 1
  - equal:
      path: spec.resources.requests.storage
      value: 16Gi

