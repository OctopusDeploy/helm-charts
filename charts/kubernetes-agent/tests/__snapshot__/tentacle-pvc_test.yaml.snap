should match snapshot when storageClassName is set:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: octopus-agent
        app.kubernetes.io/version: 8.1.1494
        helm.sh/chart: kubernetes-agent-0.10.0
      name: octopus-agent-RELEASE-NAME-pvc
    spec:
      accessModes:
        - ReadWriteMany
      resources:
        requests:
          storage: 10Gi
      storageClassName: my-storage-class
